### PR TITLE
RavenDB-14570 tighten ReAddMemberNode test

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Commercial/LicenseStatus.cs [hunk 1]
```diff
@@ -70,9 +70,11 @@ public bool Expired
             }
         }
 
-        public int MaxCores => GetValue<int?>("cores") ?? 3;
+        internal static bool UnderTest;
 
-        public int MaxMemory => GetValue<int?>("memory") ?? 6;
+        public int MaxCores => GetValue<int?>("cores") ?? (UnderTest ? 3 : 256);
```
Comment (ppekrol) [inline src/Raven.Server/Commercial/LicenseStatus.cs]: What is this?

Comment (karmeli87) [reply src/Raven.Server/Commercial/LicenseStatus.cs]: Should be the other way around.. (UnderTest ? 256 : 3)
If we unable to set the license we should use 256 cores as default and not 3

File: src/Raven.Server/Rachis/Leader.cs [hunk 2]
```diff
@@ -57,6 +57,8 @@ public class Leader : IDisposable
         private readonly ConcurrentDictionary<string, FollowerAmbassador> _nonVoters =
             new ConcurrentDictionary<string, FollowerAmbassador>(StringComparer.OrdinalIgnoreCase);
 
+        public Dictionary<string, FollowerAmbassador> CurrentPeers => new  Dictionary<string, FollowerAmbassador>(_voters.Concat(_nonVoters).Concat(_promotables));
```
Comment (ayende) [inline src/Raven.Server/Rachis/Leader.cs]: What updates this? What happens when we add a new node? Remove one?

Comment (karmeli87) [reply src/Raven.Server/Rachis/Leader.cs]: `RefreshAmbassadors` will update this and handle add/remove.

Comment (ayende) [reply src/Raven.Server/Rachis/Leader.cs]: This is only called from the tests, right?

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Rachis/Leader.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Rachis/Leader.cs]: Yes, but maybe we should add a comment 'DEBUG ONLY', just in case

**Pull request merged**